WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift

```diff
@@ -0,0 +1,187 @@
+///
+final class CardReaderSettingsConnectedViewController: UIViewController {
+
+    /// Main TableView
+    ///
+    @IBOutlet weak private var tableView: UITableView!
+
+    /// ViewModel
+    ///
+    private var viewModel: CardReaderSettingsConnectedViewModel?
+
+    /// Table Sections to be rendered
+    ///
+    private var sections = [Section]()
+
+    // MARK: - Overridden Methods
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        registerTableViewCells()
+        configureNavigation()
+        configureSections()
+        configureTable()
+    }
+
+    func configure(viewModel: CardReaderSettingsConnectedViewModel) {
+        self.viewModel = viewModel
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension CardReaderSettingsConnectedViewController {
+
+    /// Set the title and back button.
+    ///
+    func configureNavigation() {
+        title = Localization.title
+    }
+
+    /// Setup the sections in this table view
+    ///
+    func configureSections() {
+        sections = [Section(title: nil,
+                            rows: [
+                                .temporaryHeader
+                            ])]
+    }
+
+    func configureTable() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.reloadData()
+    }
+
+    /// Register table cells.
+    ///
+    func registerTableViewCells() {
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+    }
+
+    /// Cells currently configured in the order they appear on screen
+    ///
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as TitleTableViewCell where row == .temporaryHeader:
+            configureHeader(cell: cell)
+        default:
+            fatalError()
+        }
+    }
+
+    private func configureHeader(cell: TitleTableViewCell) {
+        cell.titleLabel?.text = Localization.temporaryHeader
+        cell.selectionStyle = .none
+    }
+}
+
+// MARK: - Convenience Methods
+//
+private extension CardReaderSettingsConnectedViewController {
+
+    func rowAtIndexPath(_ indexPath: IndexPath) -> Row {
+        return sections[indexPath.section].rows[indexPath.row]
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension CardReaderSettingsConnectedViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
```

Comment:
Nice aussi. Since both header and footer support autosizing, would you consider doing the same for the cells? 

Wherever you configure the table view, you could set `        tableView.rowHeight = UITableView.automaticDimension`

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/4144/commits/7fc5fb81612016e63aa14cb6626d054dce13b96a